python/src/deltachat/testplugin.py

```diff
@@ -178,6 +179,50 @@ def get_liveconfig_producer(self):
                     yield config
             pytest.fail("more than {} live accounts requested.".format(MAX_LIVE_CREATED_ACCOUNTS))
 
+    def cache_maybe_retrieve_configured_db_files(self, cache_addr, db_target_path):
+        db_target_path = py.path.local(db_target_path)
+        assert not db_target_path.exists()
+
+        print("checking cache for", cache_addr)
+        try:
+            filescache = self._addr2files[cache_addr]
+        except KeyError:
+            print("CACHE FAIL for", cache_addr)
+            return False
+        else:
+            print("CACHE HIT for", cache_addr)
+            targetdir = db_target_path.dirpath()
+            write_dict_to_dir(filescache, targetdir)
+            return True
+
+    def cache_maybe_store_configured_db_files(self, acc):
+        addr = acc.get_config("addr")
+        assert acc.is_configured()
+        # don't overwrite existing entries
+        if addr not in self._addr2files:
+            print("storing cache for", addr)
+            basedir = py.path.local(acc.get_blobdir()).dirpath()
+            self._addr2files[addr] = create_dict_from_files_in_path(basedir)
+            return True
+
+
+def create_dict_from_files_in_path(path):
+    base = py.path.local(path)
+    cachedict = {}
+    for path in base.visit(fil=py.path.local.isfile):
+        cachedict[path.relto(base)] = path.read_binary()
+    return cachedict
+
+
+def write_dict_to_dir(dic, target_dir):
```

Comment:
> Why not just store a single file with [JSON](https://docs.python.org/3/library/json.html) or [pickle](https://docs.python.org/3/library/pickle.html)? This is like 1 lines instead of going through files in a directory:
> 
> ```
> from pathlib import Path
> with Path(target_dir).open('w') as fp: json.dump(dic, fp)
> ```

There is no persistence for the caching, on purpose, and to keep it simple. Each testprocess just maintains an in-process cache so we don't need to care for outdated accounts (CI email accounts are valid for one hour). 
Note that the code transfers a directory of files into a dict, and from a dict to a directory.  It's a cheap way to record and replay a DC database dir with blob files and "shm" and "wal" files etc.